### PR TITLE
New Swagger specification for Microsoft.SqlVirtualMachine Resource Provider. 

### PR DESCRIPTION
**Pull request by yareyes**
_The SQL IaaS Team is adding a new Resource Provider. This Swagger contains all the restful operations that can be performed by the customers._ 

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by yareyes** [comment]
Adding @jaredmoo.

**Comment by jaredmoo** [comment]
You will need a readme.md. I can help you with this.

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 1]
```diff
@@ -0,0 +1,1868 @@
...
+      "description": "Display metadata associated with the operation.",
+      "type": "object",
+      "properties": {
+        "provider": {
+          "description": "The localized friendly form of the resource provider name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resource": {
+          "description": "The localized friendly form of the resource type related to this action/operation.",
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: `SqlVirtualMachineGroupProperties`

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: `Provisioning State` is not a product name, so it should be `Provisioning state`.

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: "WSFC" is a lot of letters to mentally process, can the description explain these? :)

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: SqlVirtualMachineGroupListResult - the list type name must match the item type name plus "ListResult"

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Type names shouldn't have spaces in them. `SqlVirtualMachineGroupProperties`

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Nice description!!

**Review by jaredmoo — CHANGES_REQUESTED** [review]
I am being critical of some naming & description, but you have done an awesome job here so far!

**Comment by yareyes** [comment]
@pratraw is added to the review. #Closed

**Comment by yareyes** [comment]
@seliu is added to the review. #Closed

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 2]
```diff
@@ -0,0 +1,1832 @@
...
+          "Gets an availability group listener.": {
+            "$ref": "./examples/GetAvailabilityGroupListener.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "AvailabilityGroupListeners"
+        ],
+        "description": "Creates or Updates an availability group listener.",
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: `updates`

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 3]
```diff
@@ -0,0 +1,1832 @@
...
+          "Gets a SQL virtual machine group.": {
+            "$ref": "./examples/GetSqlVirtualMachineGroup.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "SqlVirtualMachineGroups"
+        ],
+        "description": "Creates or Updates a SQL virtual machine group.",
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: `updates`

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 4]
```diff
@@ -0,0 +1,1832 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachineGroups": {
+      "get": {
+        "tags": [
+          "SqlVirtualMachineGroups"
+        ],
+        "description": "Gets all the SQL virtual machine groups in a resource group.",
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: It's debatable, but I would remove "the". "Gets all SQL virtual machine groups in a resource group" sounds better to me.

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 5]
```diff
@@ -0,0 +1,1832 @@
...
+          "Gets a SQL virtual machine.": {
+            "$ref": "./examples/GetSqlVirtualMachine.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "SqlVirtualMachines"
+        ],
+        "description": "Creates or Updates a SQL virtual machine.",
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: `updates`

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 6]
```diff
@@ -0,0 +1,1832 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.SqlVirtualMachine/sqlVirtualMachines": {
+      "get": {
+        "tags": [
+          "SqlVirtualMachines"
+        ],
+        "description": "Gets all the SQL virtual machines in a subscription.",
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Same comment about "the"

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 7]
```diff
@@ -0,0 +1,1832 @@
...
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Operation": {
+      "description": "SQL REST API operation definition.",
+      "type": "object",
+      "properties": {
+
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Is "ReadOnly" missing from this enum?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: What do example values look like? Is it an enum?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: WSFC -> Wsfc ?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Spell out what WSFC is?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: What is OU?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Nice description explaining an important constraint on this property!

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Is this really just "update" and not "create"/"update"?

Comment (jaredmoo) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Update-only is pretty uncommon

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Same question about update-only

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Example values or enum?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Update-only?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: WSFC -> Wsfc ?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Can you explain this a little more? What is auto telemetry and how is region relevant?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: enum?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Which time zone?

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaredmoo) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Also "conectivity" is a typo

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json (hunk 7 above)
Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Added what OU means :)

Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Will add examples in the description.

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Yes, adding more comments on what each property means.

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json (hunk 7 above)
Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Added, local VM time.

Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Done :)

Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Removing property, not relevant.

Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Yes, group has to be created beforehand.

Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Done :)

Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Changed the description.

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json [hunk 8]
```diff
@@ -0,0 +1,1792 @@
...
+          "readOnly": true
+        }
+      }
+    },
+    "Operation": {
+      "description": "SQL REST API operation definition.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The
```
Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: The example is very helpful, thanks

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: This matches SQL's ResourceIdentity definition, but doesn't match ARM's Identity definition ( https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json#L216 ). SQL's can't be changed now without breaking change. I'm fine with it as-is, just fyi.

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: Are these passwords non-readable, i.e. mutability = create,update ?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: What are min & max values?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: measured in what?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: in what units?

Comment (jaredmoo) [inline specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: I don't fully understand this description

**Review by jaredmoo — APPROVED** [review]

**Comment by sergey-shandar** [comment]
@yareyes could you address comments from @jaredmoo ?

File: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json (hunk 8 above)
Comment (yareyes) [reply specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2017-03-01-preview/sqlvm.json]: We will keep SQL Resource Identity.

**Comment by yareyes** [comment]
@sergey-shandar comments have been addressed.

**Comment by jaredmoo** [comment]
The description updates are great, thanks!

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**